### PR TITLE
Template management - Template Creation Front End

### PR DESCRIPTION
**Pull request by soujash-mandal**
Front end Part Of Template Management
1. added routes of custom template form
2. make page containing form for template creation (added 1 file)
3. make template form component (added two files)
4. replaces switch cases with loops
5. added new algolia index named template to store templates

Screenshot 2023-07-11 at 2 22 47 AM

### CODE REVIEW

File: web/app/components/new/template-form.hbs [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{{! @glint-nocheck: not typesafe yet }}
+{{!-- custom-template-add --}}
+{{#if this.docIsBeingCreated}}
```
Comment (yashcode00) [inline web/app/components/new/template-form.hbs:3]: can these variables be renamed?

File: web/app/components/new/template-form.hbs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            </F.HelperText>
+          </Hds::Form::Textarea::Field>
+      </div>
+
+      {{!-- doc id  --}}
+      <div class="pt-10 space-y-7">
+        <Hds::Form::TextInput::Field
+          @type="text"
+          @isRequired={{true}}
+          @value={{this.docId}}
```
Comment (yashcode00) [inline web/app/components/new/template-form.hbs:81]: can these variables be renamed? (this.docid)

File: web/app/components/new/template-form.hbs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        <Hds::Form::TextInput::Field
+          @type="text"
+          @isRequired={{true}}
+          @value={{this.templateId}}
+          name="templateId"
+          placeholder="Enter new Template Id"
+          {{on "input" this.updateForm}}
+          as |F|
+        >
+          <F.Label>Template ID</F.Label>
```
Comment (yashcode00) [inline web/app/components/new/template-form.hbs:35]: shall it be template name?

File: web/app/components/new/template-form.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        .then((response) => response?.json());
+
+      // Wait for document to be available.
+      await timeout(AWAIT_DOC_DELAY);
+
+      // Set modal on a delay so it appears after transition.
+      this.modalAlerts.setActive.perform(
+        "docCreated",
+        AWAIT_DOC_CREATED_MODAL_DELAY
+      );
```
Comment (yashcode00) [inline web/app/components/new/template-form.ts:198]: is the modal modified?

File: web/app/components/new/template-form.ts [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+      this.router.transitionTo("authenticated.new");
+    } catch (err: unknown) {
+      this.docIsBeingCreated = false;
+      this.flashMessages.add({
```
Comment (yashcode00) [inline web/app/components/new/template-form.ts:203]: need to rename this.docIsBeingCreated

File: web/app/components/new/template-form.ts [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    }
+  }
+
+  /**
+   * Creates a document draft, then redirects to the document.
+   * On error, show a flashMessage and allow users to try again.
+   */
+  private createDoc = task(async () => {
+    this.docIsBeingCreated = true;
+
```
Comment (yashcode00) [inline web/app/components/new/template-form.ts:176]: need to rename this.docIsBeingCreated

File: web/app/config/environment.d.ts [hunk 7]
```diff
@@ -14,6 +14,7 @@ declare const config: {
     apiKey: string;
     docsIndexName: string;
     draftsIndexName: string;
+    templateIndexName: string,
```
Comment (yashcode00) [inline web/app/config/environment.d.ts:17]: why is this needed?

File: web/app/routes/authenticated/new/doc.js [hunk 8]
```diff
@@ -15,21 +15,53 @@ export default class AuthenticatedNewDocRoute extends Route {
...
+      let check=true
+      data.forEach(element => {
+        if (element.templateId==params.docType) {
+          check=false
+        }
+      });
+      if(check)
+      {
+          this.flashMessages.add({
           message: `Invalid document type: ${params.docType}`,
```
Comment (yashcode00) [inline web/app/routes/authenticated/new/doc.js:52]: does this check whether the doctype in url actually exists or not?

File: web/app/services/config.ts [hunk 9]
```diff
@@ -8,6 +8,7 @@ export default class ConfigService extends Service {
   config = {
     algolia_docs_index_name: config.algolia.docsIndexName,
     algolia_drafts_index_name: config.algolia.draftsIndexName,
+    algolia_template_index_name: config.algolia.templateIndexName,
```
Comment (yashcode00) [inline web/app/services/config.ts:11]: is this fetching the index name that is mentioned by the user in the config?

File: web/app/templates/authenticated/new/index.hbs [hunk 10]
```diff
@@ -8,18 +8,17 @@
       <LinkTo
         class="w-full h-full no-underline"
         @route="authenticated.new.doc"
-        @query={{hash docType=docType.name}}
+        @query={{hash docType=docType.templateId}}
       >
```
Comment (yashcode00) [inline web/app/templates/authenticated/new/index.hbs:12]: naming convention for templateid

File: web/config/environment.js [hunk 11]
```diff
@@ -45,6 +45,7 @@ module.exports = function (environment) {
       appID: getEnv("ALGOLIA_APP_ID"),
       docsIndexName: getEnv("ALGOLIA_DOCS_INDEX_NAME", "docs"),
       draftsIndexName: getEnv("ALGOLIA_DRAFTS_INDEX_NAME", "drafts"),
+      templateIndexName: getEnv("ALGOLIA_TEMPLATE_INDEX_NAME","template"),
```
Comment (yashcode00) [inline web/config/environment.js:48]: have you added an entry in `config.hcl` to fetch `"ALGOLIA_TEMPLATE_INDEX_NAME` or added any relevant environmental variable, and what is the default for it?

**Review by yashcode00 — COMMENTED** [review]
Please try to change these small things.

File: web/app/components/new/template-form.hbs (hunk 1 above)
Comment (soujash-mandal) [reply web/app/components/new/template-form.hbs:3]: yes . We will discuss together and change all the variables to their expected names

File: web/app/components/new/template-form.hbs [hunk 12]
```diff
@@ -0,0 +1,104 @@
+{{! @glint-nocheck: not typesafe yet }}
```
Comment (soujash-mandal) [inline web/app/components/new/template-form.hbs:1]: I have copied from doc-form.hbs to maintain the same UI experience and then added my required fields
1. Template ID 
2. Template long name
3. Description
4. Document link

File: web/app/components/new/template-form.hbs (hunk 3 above)
Comment (soujash-mandal) [reply web/app/components/new/template-form.hbs:35]: yeah thanks template name will be better I will change
